runtime/src/main/java/org/corfudb/protocols/service/CorfuProtocolLayout.java

```diff
@@ -0,0 +1,116 @@
+package org.corfudb.protocols.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.protocols.wireprotocol.LayoutPrepareResponse;
+import org.corfudb.runtime.proto.service.CorfuMessage.RequestPayloadMsg;
+import org.corfudb.runtime.proto.service.CorfuMessage.ResponsePayloadMsg;
+import org.corfudb.runtime.proto.service.Layout.BootstrapLayoutRequestMsg;
+import org.corfudb.runtime.proto.service.Layout.BootstrapLayoutResponseMsg;
+import org.corfudb.runtime.proto.service.Layout.CommitLayoutRequestMsg;
+import org.corfudb.runtime.proto.service.Layout.CommitLayoutResponseMsg;
+import org.corfudb.runtime.proto.service.Layout.LayoutRequestMsg;
+import org.corfudb.runtime.proto.service.Layout.LayoutResponseMsg;
+import org.corfudb.runtime.proto.service.Layout.PrepareLayoutRequestMsg;
+import org.corfudb.runtime.proto.service.Layout.PrepareLayoutResponseMsg;
+import org.corfudb.runtime.proto.service.Layout.ProposeLayoutRequestMsg;
+import org.corfudb.runtime.proto.service.Layout.ProposeLayoutResponseMsg;
+import org.corfudb.runtime.view.Layout;
+
+import static org.corfudb.protocols.CorfuProtocolCommon.getLayout;
+import static org.corfudb.protocols.CorfuProtocolCommon.getLayoutMsg;
+
+@Slf4j
+public class CorfuProtocolLayout {
+    public static RequestPayloadMsg getLayoutRequestMsg(long epoch) {
+        return RequestPayloadMsg.newBuilder()
+                .setLayoutRequest(LayoutRequestMsg.newBuilder()
+                        .setEpoch(epoch)
+                        .build())
+                .build();
+    }
+
+    public static ResponsePayloadMsg getLayoutResponseMsg(Layout layout) {
+        return ResponsePayloadMsg.newBuilder()
+                .setLayoutResponse(LayoutResponseMsg.newBuilder()
+                        .setLayout(getLayoutMsg(layout))
+                        .build())
+                .build();
+    }
+
+    public static RequestPayloadMsg getPrepareLayoutRequestMsg(long epoch, long rank) {
+        return RequestPayloadMsg.newBuilder()
+                .setPrepareLayoutRequest(PrepareLayoutRequestMsg.newBuilder()
+                        .setEpoch(epoch)
+                        .setRank(rank)
+                        .build())
+                .build();
+    }
+
+    public static ResponsePayloadMsg getPrepareLayoutResponseMsg(PrepareLayoutResponseMsg.Type type,
+                                                                 long rank, Layout layout) {
+        return ResponsePayloadMsg.newBuilder()
+                .setPrepareLayoutResponse(PrepareLayoutResponseMsg.newBuilder()
+                        .setRespType(type)
+                        .setRank(rank)
+                        .setLayout(getLayoutMsg(layout))
+                        .build())
+                .build();
+    }
+
+    public static LayoutPrepareResponse getLayoutPrepareResponse(PrepareLayoutResponseMsg msg) {
+        return new LayoutPrepareResponse(msg.getRank(), getLayout(msg.getLayout()));
```

Comment:
Not used?

Reply:
Good catch. It should be used in the LayoutHandler when returning the response back to the caller.